files/virt_volume.sh

```diff
@@ -17,82 +17,119 @@
 # Ensure that a libvirt volume exists, optionally uploading an image.
 # On success, output a JSON object with a 'changed' item.
 
-if [[ $# -ne 4 ]] && [[ $# -ne 5 ]]; then
-    echo "Usage: $0 <name> <pool> <capacity> <format> [<image>]"
+# Parse options
+OPTIND=1
+
+if [[ $# -gt 4 ]] && [[ $# -lt 6 ]]; then
```

Comment:
Shouldn't these be the other way, and or'd i.e. `-lt 4 || -gt 6`?

Reply:
Alternative would be to check for `-n $IMAGE` etc. for required arguments after parsing.

Reply:
> Shouldn't these be the other way

Yip, terrible logic - that would always evaluate to false!

Agree, alternative is a better idea. Fixing with `! [[ -n $NAME && -n $POOL && -n $CAPACITY ]]`

Reply:
By the way, decided to make -f / $FORMAT optional and default it to qcow2 if not set.